### ISSUE TITLE
get-device-spec fails when parent directory is not created with an ambiguous message

### ISSUE THREAD

**Issue by barriosnahuel** [google/bundletool#46]
**Describe the bug**
Error message can be more descriptive.

It says that the file `store/device-specs/moto-g-5-plus.json` doesn't exists but it actually fails because the `device-specs` directory doesn't exists. When the JSON file isn't there, bundletool creates it OK.

**Bundletool version(s) affected**
Version: 0.8.0

**Stacktrace**
```
[BT:0.8.0] Error: Error while writing the output file 'store/device-specs/moto-g-5-plus.json'.
java.io.UncheckedIOException: Error while writing the output file 'store/device-specs/moto-g-5-plus.json'.
        at com.android.tools.build.bundletool.commands.GetDeviceSpecCommand.writeDeviceSpecToFile(GetDeviceSpecCommand.java:173)
        at com.android.tools.build.bundletool.commands.GetDeviceSpecCommand.execute(GetDeviceSpecCommand.java:161)
        at com.android.tools.build.bundletool.BundleToolMain.main(BundleToolMain.java:83)
        at com.android.tools.build.bundletool.BundleToolMain.main(BundleToolMain.java:46)
Caused by: java.nio.file.NoSuchFileException: store/device-specs/moto-g-5-plus.json
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
        at java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:434)
        at java.nio.file.Files.newOutputStream(Files.java:216)
        at java.nio.file.Files.write(Files.java:3292)
        at com.android.tools.build.bundletool.commands.GetDeviceSpecCommand.writeDeviceSpecToFile(GetDeviceSpecCommand.java:170)
        ... 3 more
```
**To Reproduce**
Steps to reproduce the behavior.
Include the full command being run as well as, if possible, artifacts the bug can be reproduced with.

> bundletool get-device-spec --output store/device-specs/moto-g-5-plus.json

where:
- directory store already exists
- directory device-specs doesn't exists
- JSON file doesn't exists

**Expected behavior**
It would be awesome to create the whole directories tree. If that's not possible, then it would be great to read something like: "The specified output path is invalid. Is the parent directory of the JSON file already created?"

**Known workaround**
Create the parent directory manually.

**Environment:**
OS: Mac OS 10.13.6

**Comment by barriosnahuel** [comment]
@plecesne you're welcome! Here's a fix: #47

**Comment by plecesne** [comment]
Thank you very much for the contribution!
